src/ripple/app/misc/impl/ValidatorSite.cpp

```diff
@@ -203,6 +218,14 @@ ValidatorSite::makeRequest (
             ios_,
             onFetch);
     }
+    else
+    {
+        BOOST_ASSERT(resource->pUrl.scheme == "file");
+        sp = std::make_shared<detail::WorkFile>(
+            resource->pUrl.path,
+            ios_,
+            onFetchFile);
+    }
 
     work_ = sp;
```

Comment:
`work_` is a `weak_ptr`, and I need `sp` to stay alive long enough to call `run` on the next line, and I don't want to change the order. Moving here would cause it to destruct.

Reply:
Ouch, thank you for catching that.